### PR TITLE
Api structure

### PR DESCRIPTION
**Pull request by JosueRodriguezGarcia**

### CODE REVIEW

File: src/main/java/trello/api/rest/RequestFactory.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * @(#) RunCukesTest.java Copyright (c) 2019 Jala Foundation.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RequestFactory.java]: Please remove the name of class. "@(#) RunCukesTest.java"

File: src/main/java/trello/api/rest/RequestFactory.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Map;
+
+/**
+ * RequestFactory class.
+ *
+ * @author Josue Rodriguez Garcia.
+ * @version 0.0.1
+ */
+public final class RequestFactory {
+    private static Map<String, RequestManager> requestMap = new HashMap<>();
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RequestFactory.java:30]: Please add a white line above here.

File: src/main/java/trello/api/rest/RequestManager.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/*
+ * @(#) RunCukesTest.java Copyright (c) 2019 Jala Foundation.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RequestManager.java]: Please remove the name of class. "@(#) RunCukesTest.java"

File: src/main/java/trello/api/rest/RequestManager.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import io.restassured.specification.RequestSpecification;
+
+/**
+ * RequestManager class.
+ *
+ * @author Josue Rodriguez Garcia.
+ * @version 0.0.1
+ */
+public abstract class RequestManager {
+    private String endPoint;
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RequestManager.java:25]: Please add white line  above here.

File: src/main/java/trello/entities/Context.java [hunk 5]
```diff
@@ -12,24 +12,28 @@
 
 package trello.entities;
 
+
```
Comment (raulchoque-eng) [inline src/main/java/trello/entities/Context.java]: Please delete this white line.

File: src/main/java/trello/entities/Context.java [hunk 6]
```diff
@@ -67,4 +80,31 @@ public Board getBoard() {
     public List getList() {
         return list;
     }
+
+    /**
+     * Sets context's user.
```
Comment (raulchoque-eng) [inline src/main/java/trello/entities/Context.java]: Fix this comment please you set a List not user

File: src/main/java/trello/entities/Context.java [hunk 7]
```diff
@@ -67,4 +80,31 @@ public Board getBoard() {
...
+     * Gets the context's card.
+     *
+     * @return the card that is on the context.
+     */
+    public Card getCard() {
+        return card;
+    }
+
+    /**
+     * Sets context's user.
```
Comment (raulchoque-eng) [inline src/main/java/trello/entities/Context.java]: Please fix this comment you set a Card not User.

File: src/test/java/runner/RunCukesTest.java [hunk 8]
```diff
@@ -28,7 +28,7 @@
                 "html:target/cucumber",
                 "json:target/cucumber.json"},
         glue = {"trello/steps", "trello/hooks"},
-        features = {"src/test/resources/features"},
+        features = {"src/test/resources/features/login.feature"},
```
Comment (raulchoque-eng) [inline src/test/java/runner/RunCukesTest.java]: Please delete these words: "/login.feature"

File: src/test/java/trello/hooks/api/BoardAPIHook.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import trello.entities.Context;
+
+/**
+ * BoardHook class.
+ *
+ * @author Josue Rodriguez.
+ * @version 0.0.1
+ */
+public class BoardAPIHook {
+    private Context context;
```
Comment (raulchoque-eng) [inline src/test/java/trello/hooks/api/BoardAPIHook.java:31]: Please add a white line above here.

File: src/test/java/trello/hooks/api/ListAPIHook.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+import trello.entities.Context;
+
+/**
+ * ListAPIHook class.
+ *
+ * @author Josue Rodriguez.
+ * @version 0.0.1
+ */
+public class ListAPIHook {
+    private Context context;
```
Comment (raulchoque-eng) [inline src/test/java/trello/hooks/api/ListAPIHook.java:30]: Please add white line above here.

File: src/test/java/trello/hooks/api/CardAPIHook.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+import trello.entities.Context;
+
+/**
+ * CardAPIHook class.
+ *
+ * @author Josue Rodriguez.
+ * @version 0.0.1
+ */
+public class CardAPIHook {
+    private Context context;
```
Comment (raulchoque-eng) [inline src/test/java/trello/hooks/api/CardAPIHook.java:31]: Please add white line above here.

File: src/main/java/trello/api/rest/RequestFactory.java (hunk 1 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RequestFactory.java]: Solved

File: src/main/java/trello/api/rest/RequestFactory.java (hunk 2 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RequestFactory.java:30]: Solved

File: src/main/java/trello/api/rest/RequestManager.java (hunk 3 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RequestManager.java]: Solved

File: src/main/java/trello/api/rest/RequestManager.java (hunk 4 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RequestManager.java:25]: Solved

File: src/main/java/trello/entities/Context.java (hunk 5 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/entities/Context.java]: Solved

File: src/test/java/runner/RunCukesTest.java (hunk 8 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/runner/RunCukesTest.java]: Solved

File: src/main/java/trello/entities/Context.java (hunk 7 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/entities/Context.java]: Solved

File: src/main/java/trello/entities/Context.java (hunk 6 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/entities/Context.java]: Solved

File: src/test/java/trello/hooks/api/BoardAPIHook.java (hunk 9 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/trello/hooks/api/BoardAPIHook.java:31]: Solved

File: src/test/java/trello/hooks/api/ListAPIHook.java (hunk 10 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/trello/hooks/api/ListAPIHook.java:30]: Solved

File: src/test/java/trello/hooks/api/CardAPIHook.java (hunk 11 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/trello/hooks/api/CardAPIHook.java:31]: Solved

File: src/main/java/trello/api/rest/RequestFactory.java (hunk 1 above)
Comment (raulchoque-eng) [reply src/main/java/trello/api/rest/RequestFactory.java]: Review one more time please.

Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RequestFactory.java]: fix

**Review by raulchoque-eng — APPROVED** [review]

**Pull request merged**